lib/stats-collector.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * Test statistics collector.
+ *
+ * @typedef {Object} StatsCollector
+ * @property {number} suites - integer count of suites run.
+ * @property {number} tests - integer count of tests run.
+ * @property {number} passes - integer count of passing tests.
+ * @property {number} pending - integer count of pending tests.
+ * @property {number} failures - integer count of failed tests.
+ * @property {Date} start - time when testing began.
+ * @property {Date} end - time when testing concluded.
+ * @property {number} duration - number of msecs that testing took.
+ */
+
+/**
+ * Provides stats such as test duration,
+ * number of tests passed / failed etc.
+ *
+ * @public
+ * @memberof Mocha
+ * @param {Runner} runner
+ * @return {StatsCollector}
+ */
+function createStatsCollector(runner) {
+  var stats = {
+    suites: 0,
+    tests: 0,
+    passes: 0,
+    pending: 0,
+    failures: 0
+  };
+
+  if (!runner) {
+    throw new Error('Missing runner argument');
```

Comment:
TypeError. Sorry, missed this one last go-round.

Reply:
Ah ok, done.